conda_build/create_test.py

```diff
@@ -18,6 +18,59 @@
     from .metadata import MetaData
 
 
+def _normalize_path_separators_in_command(cmd: str, is_windows: bool) -> str:
+    """
+    Normalize path separators in test commands to ensure consistency.
+
+    This function detects the separator used in environment variables and matches
+    the path separators accordingly, rather than assuming Windows = backslashes.
+    Only normalizes the specific parts containing environment variables.
+    """
+    path_env_vars = [
+        "SP_DIR",
+        "PREFIX",
+        "BUILD_PREFIX",
+        "SRC_DIR",
+        "STDLIB_DIR",
+        "RECIPE_DIR",
+        "LIBRARY_PREFIX",
+        "LIBRARY_BIN",
+        "LIBRARY_INC",
+        "LIBRARY_LIB",
+        "SCRIPTS",
+        "SYS_PREFIX",
+        "ROOT",
+    ]
+
+    if not is_windows:
+        return cmd
+
+    # Find which environment variables are present in the command
+    cmd_path_env_vars = [var for var in path_env_vars if f"%{var}%" in cmd]
+
+    if not cmd_path_env_vars:
+        return cmd
+
+    # Determine the expected separator for each environment variable
+    envs_separator = {}
+    for env_var in cmd_path_env_vars:
+        env_value = os.environ.get(env_var, "")
+        envs_separator[env_var] = (
+            "/" if env_value.count("/") > env_value.count("\\") else "\\"
+        )
```

Comment:
If the variable is undefined it will always resolve to `\`

We should call that out in a comment so it is clear that is intentional and add a test for this case

Reply:
We can do a test case but I was only going to target environment variables that we know are created by conda-build.